### ISSUE TITLE
numpy.gradient inconsistent returns

### ISSUE THREAD

**Issue by kcbrowning** [numpy/numpy#7548]
```python
x = np.arange(20)
np.gradient(x)
```
returns an array. While,
```python
x = np.arange(20)
x = x.resape(10,2)
np.gradient(x)
```
returns a list of arrays. For consistency the returns should be the same or force a default axis=0 argument so that an array is always returned.

As a side note, it would be nice if gradient could handle the following...
```python
x = np.arange(20)
x = x.resape(10,2,1)
np.gradient(x)
```

> ValueError: Shape of array too small to calculate a numerical gradient, at least two elements are required.

and only fail if the user requests the gradient along the signal element axis.

**Comment by ahaldane** [comment]
Thanks for the report.

Happily, your second issue is solved in numpy 1.11, which added an `axis` parameter to `np.gradient`. See #6129. So you can do `np.gradient(np.arange(20).reshape(10,2,1), axis=(0,1))`.

As to the inconsistency in return values: I believe this is the indended behavior but the docstring is misleading. It is true that the docstring says `np.gradient` returns a "list of ndarray", but if you look lower in the examples section there is a description of how it returns an ndarray in the 1d case and a list of ndarrays otherwise.

I suggests that we should just update the docstring to reflect reality. The docstring has other problems too, see #6847.

**Comment by ahaldane** [comment]
Oh, the docstring was already changed in #5757, but not well enough :)
